src/openclsim/critical_path/base_cp.py

```diff
@@ -66,9 +70,198 @@ def _make_recorded_activities_df(self):
         """
         Set a recorded_activity_df in self.
         Uses the logs of provided activities and sim objects, combines these, adds unique UUID
-        and reshape into format such that single row has a start time and an end time
+        and reshape into format such that single row has a start time and an end time.
         """
-        pass
+        # get all recorded log-events
+        all_recorded_events = self.combine_logs()
+
+        # reshape into set of activities with start, duration and end
+        recorded_activities_df = self.reshape_log(all_recorded_events)
+
+        # add unique identifier for activities (could be shared by multiple objects)
+        self.recorded_activities_df = self.add_unique_activity(recorded_activities_df)
+
+    def combine_logs(self):
+        """
+        Combines the logs of given (simulation) objects into a single dataframe.
+        """
+        # check unique names - @Pieter maybe warning?
+        names = [obj.name for obj in self.object_list]
+        assert len(names) == len(set(names)), "Names of your objects must be unique!"
+
+        # concat
+        log_all = pd.DataFrame()
+        for obj in self.object_list:
+            log = get_log_dataframe(obj)
+            log["SimulationObject"] = obj.name
+            log_all = pd.concat([log_all, log])
+
+        # now drop some columns not directly needed
+        log_all = log_all.loc[
+            :, ["Activity", "Timestamp", "ActivityState", "SimulationObject"]
+        ]
```

Comment:
why not log_all.drop(columns=["Activity", "Timestamp", "ActivityState", "SimulationObject"])